engine/jobmaster/dm/checkpoint/agent.go

```diff
@@ -119,19 +126,13 @@ func (c *AgentImpl) Create(ctx context.Context) error {
 	return nil
 }
 
-// Update implements Agent.Update
-func (c *AgentImpl) Update(ctx context.Context, cfg *config.JobCfg) error {
-	c.logger.Info("update checkpoint")
-	c.mu.Lock()
-	c.cfg = cfg
-	c.mu.Unlock()
-	return c.Create(ctx)
-}
-
 // Remove implements Agent.Remove
 func (c *AgentImpl) Remove(ctx context.Context) error {
 	c.logger.Info("remove checkpoint")
-	cfg := c.getConfig()
+	cfg, err := c.getConfig(ctx)
```

Comment:
And here we should make sure lifetime of job in jobStore should cover Agent. Maybe left comment to describe the relationship of components